### ISSUE TITLE
base: handle 'unpack_string()' error in 'Arg::string()'

### ISSUE THREAD

**Issue by cproc** [genodelabs/genode#4100]
When building `drivers/ahci` for x86_64 with gcc 10, the following error occurs:

```
  Program drivers/ahci/spec/x86/ahci_drv
    COMPILE  main.o
In file included from /.../repos/base/include/util/token.h:17,
                 from /.../repos/base/include/util/xml_node.h:18,
                 from /.../repos/base/include/base/attached_rom_dataspace.h:18,
                 from /.../repos/os/src/drivers/ahci/main.cc:15:
/.../repos/base/include/util/string.h: In member function ‘virtual Genode::Session_capability Ahci::Main::session(const Session_args&, const Genode::Affinity&)’:
/.../repos/base/include/util/string.h:82:15: error: ‘buf’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
   82 |   for (; s && *s; s++, res++);
      |               ^~
```

The error message appears somewhat misleading (there is no `buf` variable at the indicated location). At the end it turned out that the actual error location is in the `Genode::label_from_args()` function:

```
inline Session_label label_from_args(char const *args)
{
	char buf[Session_label::capacity()];
	Arg_string::find_arg(args, "label").string(buf, sizeof(buf), "");
 
	return Session_label(Cstring(buf));
}
```

The `Genode::Arg::string()` function does not initialize 'buf' with the default string if `Genode::unpack_string()` returns with an error code.

**Comment by cproc** [comment]
Fixed in 98bbf42

**Comment by chelmuth** [comment]
Merged to staging.
